pkg/iac-providers/arm/v1/load-file.go

```diff
@@ -0,0 +1,279 @@
+				_, ok := config.Config.(map[string]interface{})
+				if !ok {
+					zap.S().Debug("unable to parse config.Config data",
+						zap.String("resource", r.Type), zap.String("file", absFilePath),
+					)
+					continue
+				}
+
+				for _, nr := range r.Resources {
+					if !strings.HasPrefix(nr.Type, "Microsoft.") {
+						nr.Type = r.Type + "/" + nr.Type
+					}
+
+					resourceConfigs := a.getConfig(doc, absFilePath, m, nr, template.Variables)
+					a.addConfig(allResourcesConfig, resourceConfigs)
+				}
+			}
+			a.addConfig(allResourcesConfig, configs)
+		}
+	}
+	return allResourcesConfig, nil
+}
+
+func (ARMV1) getFileType(file string) string {
+	if ext := filepath.Ext(file); strings.EqualFold(ext, JSONExtension) {
+		return JSONExtension
+	}
+	return UnknownExtension
+}
+
+func (ARMV1) extractTemplate(doc *utils.IacDocument) (*types.Template, error) {
+
+	const errUnsupportedDoc = "unsupported document type"
+
+	if doc.Type == utils.JSONDoc {
+		var t types.Template
+		err := json.Unmarshal(doc.Data, &t)
+		if err != nil {
+			return nil, err
+		}
+		return &t, nil
+	}
+	return nil, errors.New(errUnsupportedDoc)
+}
+
+func (ARMV1) addConfig(a output.AllResourceConfigs, configs []output.ResourceConfig) {
+	for _, config := range configs {
+		if _, present := a[config.Type]; !present {
+			a[config.Type] = []output.ResourceConfig{config}
+		} else {
+			resources := a[config.Type]
+			if !output.IsConfigPresent(resources, config) {
+				a[config.Type] = append(a[config.Type], config)
+			}
+		}
+	}
+}
+
+// getSourceRelativePath fetches the relative path of file being loaded
+func (a *ARMV1) getSourceRelativePath(sourceFile string) string {
+	// rootDir should be empty when file scan was initiated by user
+	if a.absRootDir == "" {
+		return filepath.Base(sourceFile)
+	}
+	relPath, err := filepath.Rel(a.absRootDir, sourceFile)
+	if err != nil {
+		zap.S().Debug("error while getting the relative path for", zap.String("IAC file", sourceFile), zap.Error(err))
+		return sourceFile
+	}
+	return relPath
+}
+
+func (a *ARMV1) getConfig(doc *utils.IacDocument, path string, m core.Mapper, r types.Resource,
+	vars map[string]interface{}) []output.ResourceConfig {
+
+	// handle linked template
+	if r.Type == deployments {
+		var parameters map[string]struct {
+			Value interface{} `json:"value"`
+		}
+		// get linked template
+		var templateData []byte
+		var templateSource string
+		if r.Properties != nil {
+			// nested template
+			if nestedTemplate, ok := r.Properties["template"]; ok {
+				templateData, _ = json.Marshal(nestedTemplate)
+				templateSource = a.getSourceRelativePath(path)
+			} else if templateLink, ok := r.Properties["templateLink"].(map[string]interface{}); ok {
+				// download from templatelink uri
+				if uri, ok := templateLink["uri"].(string); ok {
+					tempDir := filepath.Join(os.TempDir(), utils.GenRandomString(6))
+					defer os.RemoveAll(tempDir)
+					templatePath, err := a.downloadTemplate(uri, tempDir)
```

Comment:
`uri` can also be an expression. should that be resolved before downloading?

Reply:
Linked templates removed from this PR